web-server/src/slices/service.ts

```diff
@@ -0,0 +1,76 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { ServiceNames } from '@/constants/service';
+
+type Status = {
+  isUp: boolean;
+};
+
+type Service = {
+  isUp: boolean;
+  logs: string[];
+};
+
+export type ServiceStatusState = {
+  [key in ServiceNames]: Service;
+};
+
+type State = {
+  services: ServiceStatusState;
+  loading: boolean;
+  error?: string;
+  active: string | null;
+};
+
+const initialState: State = {
+  services: {
+    [ServiceNames.API_SERVER]: { isUp: false, logs: [] },
+    [ServiceNames.REDIS]: { isUp: false, logs: [] },
+    [ServiceNames.POSTGRES]: { isUp: false, logs: [] },
+    [ServiceNames.SYNC_SERVER]: { isUp: false, logs: [] }
+  },
+  active: null,
+  loading: true,
+  error: undefined
+};
+
+type SetStatusPayload = {
+  statuses: { [key in ServiceNames]: Status };
+};
+
+export const serviceSlice = createSlice({
+  name: 'services',
+  initialState,
+  reducers: {
+    setActiveService: (state, action: PayloadAction<string | null>) => {
```

Comment:
Payload wont't pe string if you implement previous comments.
instead `ServiceNames`

Reply:
sure